cstar/tests/test_roms_marbl_example.py

```diff
@@ -0,0 +1,102 @@
+import builtins
+from pathlib import Path
+import yaml
+
+import cstar
+import pytest
+
+import cstar
+##
+from pathlib import Path
+import shutil
+from contextlib import contextmanager
+
+@pytest.fixture
+def mock_user_input():
+    """Monkeypatch wich will automatically respond "y" to any call for input"""
+    @contextmanager
+    def _mock_input(input_string):
+        original_input = builtins.input
+        def mock_input_function(_):
+            return input_string
+        builtins.input = mock_input_function
+        try:
+            yield
+        finally:
+            builtins.input = original_input
+    
+    return _mock_input
+
+
+# TODO we want to make this unnecessary, ideally by defining fixtures that create temporary directories for all of these
+## Delete output of any previous run of this script
+for oldfiles in ["local_input_files/",
+                 "local_additional_code/",
+                 "roms_marbl_example_case/",
+                 "roms_marbl_local_case/",
+                 "test_blueprint.yaml",
+                 "modified_test_blueprint.yaml",
+                 "test_blueprint_local.yaml"]:
+    
+    oldpath=Path(oldfiles)
+    if oldpath.exists():
+        if oldpath.is_dir():
+            shutil.rmtree(oldpath)
+        else:
+            oldpath.unlink()
+
+
+# TODO this assumes you are running pytest from the root directory of the cloned repo, which is fragile
+ROMS_MARBL_BASE_BLUEPRINT_PATH = './examples/cstar_blueprint_roms_marbl_example.yaml'
+
+
+@pytest.fixture
+def roms_marbl_blueprint_remote_datasets(tmp_path) -> Path:
+    """Creates blueprint yaml file in temporary directory"""
+    
+    with open(ROMS_MARBL_BASE_BLUEPRINT_PATH, 'r') as file:
+        blueprint_dict = yaml.load(file, Loader=yaml.Loader)
+        
+    blueprint_filepath = tmp_path / 'blueprint.yaml'
+
+    with open(blueprint_filepath, 'w') as file:
+        yaml.dump(blueprint_dict, file)
+
+    return blueprint_filepath
+
+
+# TODO similar fixture but the blueprint file contains paths to local files instead
+# TODO should these local files be in a given temporary directory?
```

Comment:
I'm not entirely sure what the best way to handle blueprints is, but I think we should change the current approach of modifying a blueprint to use machine-specific local filepaths.

The point of this fixture was supposed to be so that we could start with one blueprint and modify it in-memory, then write the modified blueprint back out to a temporary directory, then pass the (temporary) path to that file back into the test.

However it would be better to just have a set of pre-defined blueprint files to read from. If paths in blueprints can be relative paths, then perhaps we can avoid having to make the blueprint specific to a machine?

Reply:
> However it would be better to just have a set of pre-defined blueprint files to read from.

I'm not strictly against having two separate blueprints (one local, one remote) if that's what you mean here. However, I deliberately avoided this as these two files would differ only by a single substring throughout. Any change to the remote blueprint thus needs to be also tracked in the local version (this could be a lot if we switch all the input datasets, for example) whereas currently we only need to track the two substrings (base URL, base filepath). 

> I'm not entirely sure what the best way to handle blueprints is, but I think we should change the current approach of modifying a blueprint to use machine-specific local filepaths.

This is definitely fine for testing, and any user is free to make blueprints with relative paths, but I would maintain the behaviour that, within C-Star, and in any blueprints it writes out, absolute paths are used (to maintain the most exact information available about the location of a file). C-Star-produced blueprints should not break when moved from one directory to another because of the use of relative paths. Thus if we want to use relative paths in a blueprint describing local files _produced by C-Star_, we should have to manually modify it.

Reply:
I just stumbled on this discussion #46 which is similar. I forgot another reason for recycling the same blueprint was to test C-Star's ability to create a case from a C-Star-created blueprint. This is something we should definitely test if we aim to prepare separate local/remote blueprints. It may be worth planning around that this will soon double to local/remote x netcdf/yaml

Reply:
Thanks for explaining, this makes a lot more sense to me now. Keeping a single yaml dict in memory then modifying it between tests is a very clean way to handle it, nice.

Reply:
This generation and modification of the yaml as a dict in-memory is now implemented.